### PR TITLE
Fix SABnzbd single-file post-processing path remapping

### LINKED ISSUE #540: Post-processing fails when SABnzbd reports a file path instead of a folder path (single-file releases)
Summary

For SABnzbd downloads, post-processing fails with "Source path not found at Shelfarr's configured download mount" when the release contains a single file. It appears SABnzbd reports the completed download's storage path differently depending on whether the release has one file or several — a folder path for multi-file releases, but the direct file path for single-file releases. Shelfarr's post-processing step seems to always treat the reported path as a folder, so when it's actually a file, nothing is found "inside" it and the request gets flagged for attention.

Evidence

Two downloads from the same SABnzbd category ("books"), same indexer/client, same everything except file count in the release:

Worked (multi-file release, path reported as a folder):

Download #18
Path: /Data/Downloads/Usenet/Books/Suzanne Collins - The Hunger Games

Failed (single-file release, path reported as the file itself):

Download #23
Path: /Data/Downloads/Usenet/Books/Cassandra Clare - City of Ashes/Cassandra Clare - [Mortal Instruments 02] - City of Ashes (retail) (epub).epub

In the failing case, the file genuinely exists on disk at that exact path — confirmed directly via ls. Retrying the request 10+ times does not resolve it, since the path being looked up is consistently wrong regardless of retries.

Steps to reproduce
Configure a SABnzbd download client in Shelfarr.
Submit a request that resolves to a single-file NZB release (no accompanying NFO/sample/cover — just one file).
Let it download and complete via SABnzbd.
Observe post-processing fails with "Source path not found at Shelfarr's configured download mount," even though the file is present on disk.
Compare against a request that resolves to a multi-file release — this completes successfully.
Expected behavior

Post-processing should handle both cases — if the reported path is a file rather than a directory, Shelfarr should treat its parent directory as the source folder (or operate on the file directly), rather than assuming it's always a folder. This is a well-known SABnzbd integration edge case; Sonarr/Radarr handle it defensively for the same reason.

Related issue (possibly same root cause)

Separately, on a different single-file-release test, having both a .epub and a .mobi in the same completed folder caused the import to fail outright until the .mobi was manually deleted, leaving only the .epub. This may be a related gap in the same file-resolution step (no clear handling for multiple candidate files in one folder).

Environment
Shelfarr image: ghcr.io/pedro-revez-silva/shelfarr:latest (2026.09.10.1)
Download client image: linuxserver/SABnzbd:latest (5.1.3)
Deployment: Docker, Synology Container Manager

### PR DESCRIPTION
**Pull request by Pedro-Revez-Silva**
## Assigned issue

Closes #540

## What changed

SABnzbd reports a completed single-file release as the file path, while multi-file releases report the job folder. Post-processing remapping used only `File.basename`, so category/local-path lookup looked for the ebook filename at the download mount and failed with the configured-mount "source path not found" attention state even when the file was present under the job folder.

When the reported path looks like a completed download file, remapping also maps its parent job folder and appends the original filename. It never imports the parent directory in place of a missing reported file. File detection reuses the supported audiobook extension list. Parents that are a shared download root or the client category are skipped so a file sitting directly in the category folder cannot widen the import to the whole queue.

If the remapped path already exists as a file on disk, import continues to operate on that file (existing single-file handling).

## Verification

Added supported-audio-format remapping and missing-file/wrong-sibling import regressions. Post-processing, monitor, and SABnzbd suites: 201 runs, 846 assertions, zero failures. Changed-file RuboCop clean.

- `RAILS_ENV=test bundle exec ruby -Itest test/jobs/post_processing_job_test.rb` — 148 runs, 0 failures
- `RAILS_ENV=test bundle exec ruby -Itest test/jobs/download_monitor_job_test.rb` — 39 runs, 0 failures
- `RAILS_ENV=test bundle exec ruby -Itest test/services/download_clients/sabnzbd_test.rb` — 12 runs, 0 failures
- `bundle exec rubocop` on the three changed files — no offenses

Covered cases:
- Host-style SABnzbd file path remapped through `download_local_path` + category `books` through the job folder to the exact reported file, then imported
- Same-filesystem completed file path still imported directly
- Parent remapping does not fall back to the shared category folder
- Mocked SABnzbd history stores a file `storage` path unchanged

Not reproduced 1:1: Synology Container Manager with `linuxserver/SABnzbd` 5.1.3 and host path `/Data/Downloads/Usenet/Books/...`. The tests mock that layout rather than booting SABnzbd.

Out of scope: a completed folder containing both `.epub` and `.mobi` that fails until one file is removed. Directory import already accepts both ebook extensions when content is valid; that report does not share this remapping bug.

## Screenshots or recordings

Not applicable (backend remapping / post-processing only).

## Checklist

- [x] The maintainer assigned the linked issue to me before I started implementation
- [x] This pull request stays within the assigned issue's scope
- [x] I added or updated tests for the behavior I changed
- [x] I ran the relevant local quality checks
- [x] I updated documentation for user-visible changes (none required)

<div><a href="https://cursor.com/agents/bc-508a2a81-114f-537c-a3a1-415cf46ad1d4?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-508a2a81-114f-537c-a3a1-415cf46ad1d4&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: app/jobs/post_processing_job.rb [hunk 1]
```diff
@@ -1452,6 +1457,8 @@ def remap_download_path(path, download)
     Rails.logger.info "[PostProcessingJob] Resolving the download client path"
 
     candidates = build_path_candidates(path, download)
+    parent_path = parent_directory_for_file_source(path, download)
+    candidates.concat(build_path_candidates(parent_path, download)) if parent_path.present?
```
Comment (Pedro-Revez-Silva) [inline app/jobs/post_processing_job.rb]: [P2] Preserve the reported filename when remapping its parent

When SABnzbd reports `/host/books/Book job/Missing Book.epub` but the mapped job directory contains only `Other Book.epub`, these new candidates accept the entire parent directory. Reproducing through `PostProcessingJob.perform_now` imports the other ebook and marks the requested book complete; the same case correctly defers against the base implementation. Map the parent and then append the original filename, so a missing reported file cannot widen the import to unrelated sibling content.

**Review by Pedro-Revez-Silva — COMMENTED** [review]
Changes required before merge at head a65e90005dc53d7ccc0281ebb8ee5f90cac16ccf, merge-base e6f3071650e6dd1060c6ddbd33e52207c76f2e92. The PR addresses #540 and its existing suites pass, but an adversarial filesystem reproduction demonstrates wrong-file import via the new parent fallback. The existing missing-audiobook-extension finding is also confirmed and is not duplicated below. I will fix these as authorized and revalidate.

Recorded as COMMENT because GitHub prohibits the authenticated PR author from requesting changes on their own PR.

**Review by Pedro-Revez-Silva — COMMENTED** [review]
Ready to merge after fixing both verified findings. Reviewed head 48e40d7432ab8252ad2f7380b3bd8758f04da0ef against merge-base e6f3071650e6dd1060c6ddbd33e52207c76f2e92. The issue #540 host-only single-file path now resolves to the exact mounted file. The supported audiobook list is reused, and remapping never substitutes an entire parent directory for a missing reported file. Both defects were reproduced before the fix and the implementation was independently re-reviewed; no findings remain.

Validation: post-processing, monitor and SABnzbd suites 201 tests/846 assertions pass; changed-file RuboCop clean; combined bin/quality push passes including system tests and coverage gates; combined focused suite 560/2167 passes. Current-head CI quality, security, companion, both filesystem architectures, container and CodeQL all pass. Source changes include regression tests, and all relevant threads are resolved.

Recorded as COMMENT because GitHub does not permit the authenticated PR author to self-approve.
